### ISSUE TITLE
Some Copilot completions want suggest an extra unwanted indent

### ISSUE THREAD

**Issue by ryankhart** [zerolfx/copilot.el#92]
Forgive me if this is a duplicate of another issue or a problem with my own private config. I don't believe that it is a problem with my private config because I have since gutted my Doom Emacs `config.el` file a few different times already to isolate a few other issues I've been having. Regardless, this issue has persisted, so I that's why I feel like this might be a bug with this Copilot package.

Below is a screenshot of the issue I've been having. I'd copy/paste it into a code block if the bug didn't involve overlays, so this is the best way I can think of to describe this issue.

<img width="740" alt="image" src="https://user-images.githubusercontent.com/3673962/220214839-a9e629b5-5969-4ecc-99ae-90243da0970c.png">

I don't have a problem with _what_ is being suggested but the fact that it seems to want to add that extra indent when the typing cursor (`point`) is already at the indented point. This package consistently seems to add in a 6-space indent.

I've included `(setq tab-width 2)` in my `config.el` file, and the bug happens the same regardless.

I'm open to suggestions for workarounds to add to my config as well.

**Comment by ryankhart** [comment]
Ok I've added some `message` statements to echo out what's being done at the moment of this copilot completion suggestion overlay, and it seems to be reverting to the default tab width (which it seems to think is 8 spaces). This also seems to contradict my statement earlier that it was 6 spaces, but when I accepted the completion, I counted a total indent of 16 spaces (8 spaces * 2 levels deep (maybe?)).

```Elisp
(defun copilot--infer-indentation-offset ()
  "Infer indentation offset."
  (message "copilot--infer-indentation-offset entered, major mode is %s" (symbol-name major-mode))
  (or (let ((mode major-mode))
        (while (and (not (assq mode copilot--indentation-alist))
                    (setq mode (get mode 'derived-mode-parent))))
        (message "infer-indentation-offset: derived mode is %s" (symbol-name mode))
        (when mode
          (cl-some (lambda (s)
                     (when (boundp s)
                       (progn
                         (message "infer-indentation-offset: checking symbol %s" (symbol-name s))
                         (symbol-value s))))
                   (alist-get mode copilot--indentation-alist))))
      (progn
        (message "infer-indentation-offset: using default tab width %d" tab-width)
        tab-width)))
```

<img width="849" alt="Screenshot 2023-02-20 at 5 11 00 PM" src="https://user-images.githubusercontent.com/3673962/220223177-7ac1b5bc-0dc4-4ca8-be29-d2431d9c9963.png">

I'm not sure where to go from here, but I'll leave this here until I or someone else can use it to find a solution.

**Comment by ryankhart** [comment]
I think I might have found a solution, but I'll need to spend some time with it in my config before I determine if it's truly fixed.

I added the following line to my Doom Emacs config.el file
```
(setq lisp-indent-offset 2)
```

Thanks to my back and forth conversation with ChatGPT on this issue, it helped me do all of this and it told me to "check if there is an entry for lisp-mode in the `copilot--indentation-alist`" and when I did, it gave me the variable `lisp-indent-offest` which was set to `nil`. And since `emacs-lisp-mode` is derived from `lisp-mode`, `emacs-lisp-mode` didn't have an offset either.

I'll spend some time with this and come back in maybe a few days depending on how much time I spend in Emacs and then report back on this.
